magento/pwa-studio
# refactor: 5x faster streaming image opto with no SSL failures

## Description

- Replace `express-sharp` with `hastily`, an Express middleware which directly simulates Fastly image optimization.
- Remove routing to image opto, since `hastily` works inline with other middlewares, modifying the response stream. Less configuration!
- Simplify `addImgOptMiddleware` a lot, since it no longer has to translate parameters AND we can deprecate some little-used features.
- Set UPWARD's `ignoreSSLErrors` configuration to respect `NODE_ENV`.

Check this out, it's super fast.

ALSO:
- Update `devcert` to include the 825-day Catalina limit.

## Related Issue





Closes #1885. Does it extra hard.
#1969 
#1311 
#1443 
#1197

## Acceptance 

### Verification Stakeholders
@magento/pwa-studio-team 

### Specification
No UPWARD spec changes, but forthcoming we might need to add custom CAs to the ProxyResolver and ServiceResolver.

### Verification Steps


1. Check out the code branch.
2. Set the MAGENTO_BACKEND_URL to the **secure (https) base url** of a local instance, or an instance with a known-bad SSL certificate.
3. Run `yarn run watch:venia`.
4. Load the homepage and click around.

Before, all the images would bomb out in a local instance because of SSL cert failures, unless you manually set very unsecure options like `NODE_TLS_REJECT_UNAUTHORIZED`. Now it doesn't, because image requests are going through the same `veniaProxy` in `packages/venia-ui/upward.yml` which already has `ignoreSSLErrors` set for the MAGENTO_BACKEND_URL origin only.


## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
